sdk/messaging/azservicebus/receiver.go

```diff
@@ -181,7 +172,13 @@ func (r *Receiver) newReceiverLink(ctx context.Context, session amqpwrap.AMQPSes
 
 // ReceiveMessagesOptions are options for the ReceiveMessages function.
 type ReceiveMessagesOptions struct {
-	// For future expansion
+	// TimeAfterFirstMessage controls how long, after a message has been received, before we return the
+	// accumulated batch of messages.
+	//
+	// Default value depends on the receive mode:
+	// - 20ms when the receiver is in ReceiveModePeekLock
+	// - 1s when the receiver is in ReceiveModeReceiveAndDelete
+	TimeAfterFirstMessage time.Duration
```

Comment:
Is this something that other SDKs have and we're just catching up?

Reply:
No, new feature we're exposing based on the idea that Go is more performance based.

There's a customer ask for it, and it's come up in an issue before about configuring the value (in that case, less rather than more).

Reply:
I see. Why are the times different depending on the mode?

Reply:
(this was awhile ago)

There was a noticeable different in the # of messages I could get, with the wait time of 20ms, between the two modes. I left it a bit longer since there's no danger of expiration so it's okay to get more messages at a time.

This "time after first message" timer is really there as a way to prevent messages from expiring, while still trying to give some perf gains by batching.